File: 20180709/FlippingAnImage.java
class Solution {
    public int invert(int x){
        if(x == 0){
            return 1;
        }
        else{
            return 0;
        }
    }
    public int[] exchange(int x, int y){
        int[] res = {y, x};
        return res;
    }
    public int[][] flipAndInvertImage(int[][] A) {
        int M = A.length;
        if(M < 1){
            int[][] no_item = {{}};
            return no_item;
        }
        for(int i = 0; i < M; i++){
            for(int j = 0; j < A[0].length / 2; j++){
                A[i][j] = invert(A[i][j]);
                A[i][A[0].length - 1 - j] = invert(A[i][A[0].length - 1 - j]);
                int[] exc = exchange(A[i][j], A[i][A[0].length - 1 - j]);
                A[i][j] = exc[0];
                A[i][A[0].length - 1 - j] = exc[1];
            }
            if(A[0].length % 2 != 0){
                    A[i][A[0].length / 2] = invert(A[i][A[0].length / 2]);
            }
        }
        return A;
    }
}

File: 20180709/FlippingAnImage.py
class Solution(object):
    def invert(self, x):
        return 0 if x == 1 else 1
    def exchange(self, a, b):
        return b,a
    def flipAndInvertImage(self, A):
        """
        :type A: List[List[int]]
        :rtype: List[List[int]]
        """
        if len(A) < 1:
            return []
        for row in A:
            for i in range(len(row)/2):
                row[i] = self.invert(row[i])
                row[len(row) - 1 - i] = self.invert(row[len(row) - 1 - i])
                row[i], row[len(row) - 1 - i] = self.exchange(row[i], row[len(row) - 1 - i])
            if (len(row)%2 != 0):
                row[len(row)/2] = self.invert(row[len(row)/2])
        return A